lib/jbuilder/pagination/pages.rb

```diff
@@ -1,29 +1,36 @@
 class Jbuilder
+  FIRST_PAGE = 1
+
   def pages!(collection, options={})
     return unless collection
 
-    pages do
-      pages_from(collection).map do |k, v|
-        _set_value k, "#{options[:url]}?page=#{v}&per_page=#{collection.per_page}"
-      end
+    pages_from(collection).map do |key, value|
+      params = query_parameters(options).merge(page: { number: value, size: collection.size }).to_query
+      _set_value key, "#{options.fetch(:url, nil)}?#{params}"
     end
   end
 
   private
 
   def pages_from(collection)
-    return {} if collection.total_pages == 1
+    return {} if collection.total_pages == FIRST_PAGE
 
     {}.tap do |pages|
-      unless collection.current_page == 1
-        pages[:first] = 1
-        pages[:prev]  = collection.current_page - 1
+      pages[:self] = collection.current_page
+
+      unless collection.current_page == FIRST_PAGE
+        pages[:first] = FIRST_PAGE
+        pages[:prev]  = collection.current_page - FIRST_PAGE
```

Comment:
Actually, `FIRST_PAGE` is 1,  You might had misunderstood because of the name. 

But I do agree with you that is quite difficult to read, so I've change the name to `ONE_PAGE`. What do you think?